### ISSUE TITLE
Major bug:  Adding small ores or altering ores causes corruption

### ISSUE THREAD

**Issue by axlegear** [Blood-Asp/GT5-Unofficial#353]
Some thorough testing has revealed a possible cause of the randomly interspersed blocks from other mods.

In a strictly IC2/GT5U environment, with both the old and new arcs, reveals that if you add custom small ores (and possibly also custom veins) that it will cause random veins or interspersed random blocks.  They seem to have a random number or something for some strange reason-  if that number comes up it picks a block from another, non-IC2/GT mod.  But sometimes it is just plain corrupt and causes a crash.
This also may occur if you alter the frequency of ore veins' generation, such as reducing/removing plutonium veins or lowering the rate of uranium.

I'm still hashing out more details but I am very confident after months of testing and probing now that it is GT, and only when the oregen is modified.  Why, and the extent of problem, is still being probed as it seems that if a world is genned and random crap appears, it will continue to occur even if the oregen is restored to defaults.  But if oregen is left default, it does not appear to ever trigger this bug... only if oregen is altered.
Also interestingly-  any added custom veins do seem to work more or less correctly.

My testing has been across the board but i've begun homing in on those which alter world-gen and systematically narrowing it down.  Finally, i've concluded without a doubt that it's GT5U....

**Comment by Blood-Asp** [comment]
Then can you supply configs, worldseed and forge version so i can test it myself? As long i can't replicate it myself, i can't do anything.

**Comment by axlegear** [comment]
Forge:  
World Seed:  -1456464890092801118

Configs seem to occur with any modification of the ore veins.  You may have to look around quite a bit because it's not 100% of the time, it's fairly rare, but very severe and can easily crash the server, cause lag, and worse.  If you need an example config I can provide one but it really works with any where you add a custom small ore, and I believe also with veins or altering spawn rates.  I added small nikolite ore at height of 1 to 5, and I also made uranium less common.

**Comment by axlegear** [comment]
Also I beleive it may be related to:

https://github.com/Blood-Asp/GT5-Unofficial/issues/235

**Comment by axlegear** [comment]
![image](https://cloud.githubusercontent.com/assets/7531754/11653158/c763ecc0-9d61-11e5-93be-669566d2c8d8.png)

**Comment by axlegear** [comment]
http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-tools/1261072-minecraft-land-generator

This mod may help generating large worlds, tho much of the time the corruption doesn't actually crash the server unless you interact with a bad block.  But it is proving reliable for causing lag and crashes on the server, while a vanilla server doesn't.

**Comment by Dream-Master** [comment]
Nice Piston Trees :)

**Comment by Blood-Asp** [comment]
In the last few days there were some reports about this chunk curruption bug on reddit and FTB forums. And as far i could see, non of the reports had Gregtech included. I will keep watching, but for me it looks like the bug is not gt related.

**Comment by axlegear** [comment]
Hopefully we can find out.  Unless it's a forge bug, tho, or IC2, I can't think of any other culprit anymore.

**Comment by LemADEC** [comment]
No cauldron on your setup?

**Comment by axlegear** [comment]
I'd never dream of it.  Just various forge builds, various IC2 builds, and various GT5 builds.  I haven't tested since .17 or so but unless something specific changed I doubt it just 'worked out'.

I've reported this in the IC2 before but they seemed to think it wasn't IC2 so, I really don't know.  Noone seems to be able to find anything wrong.

**Comment by LemADEC** [comment]
We've yet to experience world corruption on our setup.
A few of our specificites:
- we're using KCauldron.
- small ores are placed by GregTech
- normal ores in the overworld are placed by CustomOreGen
- normal ores in Nether/End are placed by GregTech
- world generation is handled by ATG + Highlands

Overworld map was pregenerated with gregtech-5.09.15 (or before).
We're currently running with gregtech-5.09.16.

**Comment by Blood-Asp** [comment]
It is also fully possible, that it gets broken on the HDD and is not even minecraft related.

**Comment by axlegear** [comment]
That was a definite suspect.  I moved the server from the HDD RAID array to a new SSD array.  However, there have been 3 different SSDs since then and the problem persists.  (2 SSDs have died, but no apparent corruption resulted.)  Also the corruption seems to occur when generating new chunks, it seems if it was a bad hardware it would affect existing chunks too.
Lastly, we have also gotten a whole new server since then as well, and it still happens.

Most of the time it seems to spawn, on the server, a wierd vein made of buildcraft frames, or trees made entirely out of redstone engines, which made me first suspect buildcraft as the problem.
But later the same problem occured in SP with a test pack that didn't have BC, and it was spawning small ores of pam's gardens underground.  So I began testing and eventually reported this.
One cause of corruption was found, but it was unrelated to this, some glitch with Minetweaker and GTTweaker.  But we discontinued use of that and the problem persisted, even in SP on a fresh world with only Forge, IC2 and GT5U.

I really don't know where the problem lies, but i've exhausted everything I can possibly think of.
Also, i've been tested for the past day, seeing if the problem occurs with the latest forge and GT5U.  I'm hoping we can, together, find the cause of the problem and get it fixed somehow.  It's rare enough to occur that it is frustratingly hard to track, but definitely persistant and recurring.  And it's so strangely specific, too!  It always happens roughly the same way.  Even with IC2+GT only, veins of blank gt ore will spawn.  But if other mods are present, there's a higher chance it will becomes random items from other mods.  I don't see any similarity either between Buildcraft Frames, Pam's Textile Gardens, or Redstone Engines.  I figured if they all shared a similar ID number or something..
I don't know!
